### PR TITLE
Bump to 0.85.0; update changelog

### PR DESCRIPTION
**Pull request by weihanglo**
[rendered](https://github.com/weihanglo/cargo/blob/version-bump/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,49 +1,229 @@
...
+- Clarify what is and isn't included by `cargo package`
+  [#14684](https://github.com/rust-lang/cargo/pull/14684)
+- Document official external commands: `cargo-clippy`, `cargo-fmt`, and `cargo-miri`.
+  [#14669](https://github.com/rust-lang/cargo/pull/14669)
+- Enhanced documentation on dependency resolution
+  [#14662](https://github.com/rust-lang/cargo/pull/14662)
+
+### Internal
+
+- Several resolver performance enhancements, including
```
Comment (epage) [inline CHANGELOG.md:39]: Perf changes are user facing, seems like putting these under Internal will hide them

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,49 +1,229 @@
...
+
 ### Changed
 
-- Enhanced the missing feature error message.
+- ❗️ Lockfile format v4 is now the default for creating/updating a lockfile.
+  Rust toolchains 1.78+ support lockfile v4.
+  For compatibility with earlier MSRV,
+  consdier setting the `package.rust-version` to 1.82 or earlier.
+  [#14595](https://github.com/rust-lang/cargo/pull/14595)
+- cargo-package: Now fails fast if the package version is already published.
```
Comment (epage) [inline CHANGELOG.md:70]: This is for cargo publish. The PR title refers to the packaging step

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,49 +1,229 @@
...
-    [#14445](https://github.com/rust-lang/cargo/pull/14445)
 - Log details of `rustc` invocation failure if no errors are seen
   [#14453](https://github.com/rust-lang/cargo/pull/14453)
-- cargo-package: Don't automatically include the current crate when packaging
-  [#14488](https://github.com/rust-lang/cargo/pull/14488)
+- Uplifted `windows-gnullvm` import libraries, aligning them with `windows-gnu`.
+  [#14451](https://github.com/rust-lang/cargo/pull/14451)
+- Suggest `cargo info` command in the `cargo search` result
+  [#14537](https://github.com/rust-lang/cargo/pull/14537)
+- Improved error reporting when a feature is not found in `activated_features`.
```
Comment (epage) [inline CHANGELOG.md:80]: Using a variable name in the public facing changes seems odd

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,49 +1,229 @@
...
+- Don't double-warn about `$CARGO_HOME/config` not having `.toml` extension.
+  [#14579](https://github.com/rust-lang/cargo/pull/14579)
+- Correct diagnostic count message when using `--message-format json`.
+  [#14598](https://github.com/rust-lang/cargo/pull/14598)
+- cargo-add: Perform fuzzy searches when translating package names
   [#13765](https://github.com/rust-lang/cargo/pull/13765)
+- cargo-new: only auto-add new packages to the workspace relative to the manifest,
+  rather than the current directory.
+  [#14505](https://github.com/rust-lang/cargo/pull/14505)
+- cargo-package: Don't automatically include the current crate when packaging
```
Comment (epage) [inline CHANGELOG.md:107]: "include" can also be read as "in the tarball". I was confused by this.

**Review by epage — COMMENTED** [review]
Feel free to r= me. I find it unlikely that I'll need to re-review.

File: CHANGELOG.md (hunk 3 above)
Comment (weihanglo) [reply CHANGELOG.md:80]: While this panic message is public facing, this seems more like an internal/nightly stuff. AFAIK only things like `-Zbindeps` hit this and the message still exposes internal debug representation.

Moved to under internals.

File: CHANGELOG.md (hunk 4 above)
Comment (weihanglo) [reply CHANGELOG.md]:
```
- ❗️ cargo-package: When using the `--package` flag, only the specified
  packages are packaged. Previously, the package in the current working
  directory was automatically selected for packaging.
```


After re-read the PR I think it justifies to have a compatibility note.

Moved to `Changed` with a big ❗️.

BTW, I didn't mention `--manifest-path` at all, although it is the package from either cwd or `--manifest-path` would be auto selected. That goes too detailed and I don't feel worthy.

**Pull request merged**